includes/class-wc-payments.php

```diff
@@ -461,6 +463,20 @@ public static function init() {
 				self::$upe_payment_gateway_map[ $payment_method->get_id() ] = new UPE_Split_Payment_Gateway( self::$api_client, self::$account, self::$customer_service, self::$token_service, self::$action_scheduler_service, $payment_method, $payment_methods, self::$failed_transaction_rate_limiter, self::$order_service );
 			}
 
+			self::$card_gateway         = self::get_payment_gateway_by_id( 'card' );
+			$card_payments_checkout     = new WC_Payments_Checkout( self::$legacy_card_gateway, self::$platform_checkout_util, self::$account, self::$customer_service );
+			self::$wc_payments_checkout = new WC_Payments_UPE_Checkout( self::get_gateway(), self::$platform_checkout_util, self::$account, self::$customer_service );
+		} elseif ( WC_Payments_Features::is_upe_deferred_intent_enabled() ) {
+			$payment_methods = [];
+			foreach ( $payment_method_classes as $payment_method_class ) {
+				$payment_method                               = new $payment_method_class( self::$token_service );
+				$payment_methods[ $payment_method->get_id() ] = $payment_method;
+			}
+			foreach ( $payment_methods as $payment_method ) {
+				self::$upe_payment_method_map[ $payment_method->get_id() ]  = $payment_method;
+				self::$upe_payment_gateway_map[ $payment_method->get_id() ] = new New_UPE_Payment_Gateway( self::$api_client, self::$account, self::$customer_service, self::$token_service, self::$action_scheduler_service, $payment_method, $payment_methods, self::$failed_transaction_rate_limiter, self::$order_service );
+			}
+
```

Comment:
I can't agree more! I remember us chatting about the UPE factory. Because of [this comment](https://github.com/Automattic/woocommerce-payments/pull/5869#discussion_r1155030491), all the changes in `class-wc-payments.php` were removed for now. Still, we need to think about the refactor here, but also keep in mind that our efforts should eventually pay off, given that there will be only one UPE implementation in the end. I'll make sure to come back to this topic with the whole team in a few days and for now, we can resolve this thread, if that sounds reasonable for you.